bugbug/models/performancebug.py

```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+
+import xgboost
+from imblearn.over_sampling import BorderlineSMOTE
+from imblearn.pipeline import Pipeline as ImblearnPipeline
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+
+from bugbug import bug_features, bugzilla, feature_cleanup, utils
+from bugbug.model import BugModel
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def _is_performance_bug(bug_data) -> bool:
+    keywords = bug_data.get("keywords", [])
+    whiteboard = bug_data.get("whiteboard", "").lower()
+    cf_performance = bug_data.get("cf_performance", "")
+
+    performance_keywords = ("perf", "topperf", "main-thread-io")
+
+    whiteboard_performance_text = (
+        "fxperf",
+        "fxperfsize",
+        "snappy",
+        "pdfjs-c-performance",
+        "pdfjs-performance",
+        "sp3",
+    )
+
+    if any(
+        keyword.startswith(performance_keyword)
+        for keyword in keywords
+        for performance_keyword in performance_keywords
+    ):
+        return True
+
+    if any(f"[{text}" in whiteboard for text in whiteboard_performance_text):
+        return True
+
+    if cf_performance and cf_performance not in ("---", "?"):
+        return True
+
+    return False
+
+
+class PerformanceBugModel(BugModel):
+    def __init__(self, lemmatization=False):
+        BugModel.__init__(self, lemmatization)
+
+        self.calculate_importance = False
+
+        feature_extractors = [
+            bug_features.HasSTR(),
+            bug_features.Severity(),
+            bug_features.Keywords(),
+            bug_features.IsCoverityIssue(),
+            bug_features.HasCrashSignature(),
+            bug_features.HasURL(),
+            bug_features.HasW3CURL(),
+            bug_features.HasGithubURL(),
+            bug_features.Whiteboard(),
+            bug_features.Patches(),
+            bug_features.Landings(),
+            bug_features.BlockedBugsNumber(),
+            bug_features.EverAffected(),
+            bug_features.AffectedThenUnaffected(),
+            bug_features.Product(),
+            bug_features.Component(),
```

Comment:
I'm curious about the way that you used to select these features.

Reply:
> I'm curious about the way that you used to select these features.

Since I wasn't exactly sure which features would yield the best performance, I opted to use the same features found in the `spambug model`, with the intention of later removing some and testing the performance once the initial state of the model is approved.

Reply:
As we discussed, features like `Patches` and `Landings` may not make sense to be included.

Reply:
I noticed that the performance impact field `cf_performance_impact` doesn't have a feature class. Would it be useful to create one?

Reply:
We could create such a feature when we need it. In the context of this model, this is not needed for the following reasons:
- The purpose of this model is to request the performance team to populate this field. If a bug already has a value assigned to this field, there is no need to utilize the model to determine whether the bug is performance-related or not.
- This field is used to label the bugs; thus, we cannot use it in the training.